src/firmware.cpp

```diff
@@ -60,118 +42,104 @@ static String pop_option(String& argument) {
   }
 }
 
-static CommandError run_help(int commandId, String argument);
+static void serialWrite(int requestID, char lineType, const String& str) {
+    if (requestID != 0) {
+        Serial.write('@');
+        Serial.print(requestID, DEC);
+        Serial.write(' ');
+    }
+
+    Serial.write(lineType);
+    Serial.write(' ');
 
-static CommandError led(int commandId, String argument) {
-  if (argument == "on") {
-    digitalWrite(LED_BUILTIN, HIGH);
-  } else if (argument == "off") {
-    digitalWrite(LED_BUILTIN, LOW);
-  } else {
-    return COMMAND_ERROR("unknown argument");
-  }
-  return OK;
+    Serial.println(str);
 }
 
-static CommandError servo(int commandId, String argument) {
-  String servoArg = pop_option(argument);
-  String widthArg = pop_option(argument);
+// The actual commands
 
-  if (argument.length() || !servoArg.length() || !widthArg.length()) {
-    return COMMAND_ERROR("servo takes exactly two arguments");
-  }
+static void readAnaloguePin(int requestID, const String& name, int pin) {
+  int reading = analogRead(pin);
+  serialWrite(requestID, '>', name + " " + String(reading));
+}
 
-  auto width = widthArg.toInt();
-  auto servo = servoArg.toInt();
-  if (servo < 0 || servo > 15) {
-    return COMMAND_ERROR("servo index out of range");
-  }
-  if (width != 0 && (width < 150 || width > 550)) {
-    return COMMAND_ERROR("width must be 0 or between 150 and 550");
-  }
-  SERVOS.setPWM(servo, 0, width);
+static CommandResponse analogueRead(int requestID, String argument) {
+  readAnaloguePin(requestID, "a0", A0);
+  readAnaloguePin(requestID, "a1", A1);
+  readAnaloguePin(requestID, "a2", A2);
+  readAnaloguePin(requestID, "a3", A3);
```

Comment:
Probably worth doing only one of these at a time as anolouge reads take a bit of time.

Reply:
My concern with this is that it will break `robot-api`, I'm not sure if we should be changing our API so much so close to Smallpeice.

Reply:
(Have spoken in person and agreed)